From 04f3cda4fdaf12e96fd6ae853755d5a83df53cfb Mon Sep 17 00:00:00 2001
From: Patrice Chotard <patrice.chotard@foss.st.com>
Date: Thu, 1 Feb 2024 17:21:08 +0100
Subject: [PATCH 61/65] fdt_support: fix fdt_copy_fixed_partitions function()

Move variable declaration at the beginning of the function.

Fixes 163c5f60ebb4 ("fdt_support: add fdt_copy_fixed_partitions function")

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git 796cdc8a053c554776b1d1396d9e95c4973fef42 fdt_support: fix fdt_copy_fixed_partitions function()]
Signed-off-by: Patrice Chotard <patrice.chotard@foss.st.com>
Change-Id: Ie9cbd3ed6e4c95b76ceb0243a5e6bfe43d68098e
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/356660
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 boot/fdt_support.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/boot/fdt_support.c b/boot/fdt_support.c
index 1c215e548db..3a384ea1470 100644
--- a/boot/fdt_support.c
+++ b/boot/fdt_support.c
@@ -1130,9 +1130,10 @@ void fdt_fixup_mtdparts(void *blob, const struct node_info *node_info,
 int fdt_copy_fixed_partitions(void *blob)
 {
 	ofnode node, subnode;
+	const u32 *reg;
 	int off, suboff, res;
 	char path[256];
-	int address_cells, size_cells;
+	int address_cells, size_cells, len;
 	u8 i, j, child_count;
 
 	node = ofnode_by_compatible(ofnode_null(), "fixed-partitions");
@@ -1178,9 +1179,6 @@ int fdt_copy_fixed_partitions(void *blob)
 			if (!ofnode_valid(subnode))
 				break;
 
-			const u32 *reg;
-			int len;
-
 			suboff = fdt_find_or_add_subnode(blob, off, ofnode_get_name(subnode));
 			res = fdt_setprop_string(blob, suboff, "label",
 						 ofnode_read_string(subnode, "label"));
-- 
2.51.0

